client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+import { bindActionCreators, Dispatch } from "redux"
+import { convertFromRaw, EditorState } from 'draft-js';
+import { gql, useQuery } from "@apollo/client";
+import React, { FunctionComponent, useState } from "react";
+import { connect } from "react-redux";
+
+import AlertDialog from '../atoms/AlertDialog'
+import { Button } from '../atoms/Button'
+import FormItem from '../molecules/FormItem'
+import FormModal from './FormModal'
+import ImagePicker from '../atoms/ImagePicker'
+import RequestGroup from '../../data/types/requestGroup'
+import RichTextField from '../atoms/RichTextField'
+import { RootState } from '../../data/reducers'
+import { TagInput } from '../atoms/TagInput'
+import { TextField } from '../atoms/TextField'
+import { upsertRequestGroup } from '../../data/actions'
+
+interface StateProps {
+  requestGroups: Array<RequestGroup>
+}
+
+interface DispatchProps {
+  upsertRequestGroup: typeof upsertRequestGroup
+}
+
+
+type Props = StateProps
+  & DispatchProps
+  & {
+    handleClose: () => void,
+    requestGroupId?: string,
+    operation: "create" | "edit"
+  };
+
+const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
+  const [initialRequestGroup, setInitialRequestGroup] = useState<RequestGroup | null>(null)
+  const [showAlertDialog, setShowAlertDialog] = useState(false)
+  const [changeMade, setChangeMade] = useState(false)
+  const [name, setName] = useState("")
+  const [description, setDescription] = useState("")
+  const [image, setImage] = useState("")
+  const [requestTypesToIdMap, setRequestTypesToIdMap] = useState(new Map<string, string>())
```

Comment:
Duplicating stuff with this state and the next line. Instead, just use an array of objects {id, name}. I think the performance hit with using array over map is preferable to maintaining two different states with duplicate information

Actually, this seems to be unused, so just remove?

Reply:
`requestTypesToIdMap` is going to be used when deciding which requestType are new and which requestTypes are deleted.

Also, `requestTypesToIdMap` is only created once when we get the requestGroup from the graphql query. not much of a performance issue

Reply:
`requestTypeNames` is the list of requestTypes that should be assigned to the request group once user submits

Reply:
This is already inside initialRequestGroup. We don't need to keep it as a state. If you do decide to keep it, change the name to reflect that it is for initialRequestTypes only

Reply:
I'm going to move `requestTypesToIdMap` to a different PR so it's usage is obvious